dbms/src/Functions/FunctionsDuration.h

```diff
@@ -88,4 +87,228 @@ class FunctionMyDurationToSec : public IFunction
+    }
+
+    bool useDefaultImplementationForConstants() const override { return true; }
+    ColumnNumbers getArgumentsThatAreAlwaysConstant() const override { return {0}; }
+
+    void executeImpl(Block & block, const ColumnNumbers & arguments, size_t result) const override
+    {
+        const auto * unit_column = checkAndGetColumnConst<ColumnString>(block.getByPosition(arguments[0]).column.get());
+        if (!unit_column)
+            throw TiFlashException(
+                fmt::format("First argument for function {} must be constant String", getName()),
+                Errors::Coprocessor::BadRequest);
+
+        String unit = Poco::toLower(unit_column->getValue<String>());
+
+        auto col_from = block.getByPosition(arguments[1]).column;
+
+        size_t rows = block.rows();
+        auto col_to = ColumnInt64::create(rows);
+        auto & vec_to = col_to->getData();
+
+        if (unit == "hour")
+            dispatch<ExtractMyDurationImpl::extractHour>(col_from, vec_to);
+        else if (unit == "minute")
+            dispatch<ExtractMyDurationImpl::extractMinute>(col_from, vec_to);
+        else if (unit == "second")
+            dispatch<ExtractMyDurationImpl::extractSecond>(col_from, vec_to);
+        else if (unit == "microsecond")
+            dispatch<ExtractMyDurationImpl::extractMicrosecond>(col_from, vec_to);
+        else if (unit == "second_microsecond")
+            dispatch<ExtractMyDurationImpl::extractSecondMicrosecond>(col_from, vec_to);
+        else if (unit == "minute_microsecond")
+            dispatch<ExtractMyDurationImpl::extractMinuteMicrosecond>(col_from, vec_to);
+        else if (unit == "minute_second")
+            dispatch<ExtractMyDurationImpl::extractMinuteSecond>(col_from, vec_to);
+        else if (unit == "hour_microsecond")
+            dispatch<ExtractMyDurationImpl::extractHourMicrosecond>(col_from, vec_to);
+        else if (unit == "hour_second")
+            dispatch<ExtractMyDurationImpl::extractHourSecond>(col_from, vec_to);
+        else if (unit == "hour_minute")
+            dispatch<ExtractMyDurationImpl::extractHourMinute>(col_from, vec_to);
+        else
+            throw TiFlashException(fmt::format("Function {} does not support '{}' unit", getName(), unit), Errors::Coprocessor::BadRequest);
+
+        block.getByPosition(result).column = std::move(col_to);
+    }
+
+private:
+    using Func = Int64 (*)(Int64);
+
+    template <Func F>
+    static void dispatch(const ColumnPtr col_from, PaddedPODArray<Int64> & vec_to)
+    {
+        if (const auto * from = checkAndGetColumn<ColumnString>(col_from.get()); from)
+        {
+            const auto & data = from->getChars();
+            const auto & offsets = from->getOffsets();
+            if (checkColumnConst<ColumnString>(from))
+            {
+                StringRef string_ref(data.data(), offsets[0] - 1);
+                constantString<F>(string_ref, from->size(), vec_to);
```

Comment:
Since `useDefaultImplementationForConstants` returns true, it looks like `col_from` should never be constant column?